### ISSUE TITLE
CMake support

### ISSUE THREAD

**Issue by virtualritz** [alexd6631/ocy#5]
Would you consider adding support for cleaning up `CMake` builds?

**Comment by alexd6631** [comment]
Hello @virtualritz, I added some basic support for CMake (ocy 0.1.4):  basically it seach for CMakeCache.txt files and delete the adjacent CMakeFiles/ directory. Let me know if it works for you.

**Comment by virtualritz** [comment]
Works like a treat, mate. This tool rocks. Sorry, but I had to tag you on Twitter and posted to HN. Hope you get lots of kudos. 🥳

**Comment by virtualritz** [comment]
Sorry, I have to reopen.

Your implementation, at the moment, only deletes the `CMakeFiles` folder. But not all the build files it references. I guess this one is not so easy. As `CMake` allows having out-of-source builds (have their own, user-created folder) as well as in-source builds (build files are scattered among sources files – eek!).

Either probably requires parsing some stuff inside the `CMakeFiles` folder to find out what exactly can be deleted.

**Comment by alexd6631** [comment]
Hey @virtualritz, indeed the current rule system in ocy is quite rudimentary right now to support this advanced use case (I plan to improve the rule system over time though, but I wanted to keep things simple for an initial release). 

I don't use CMake daily, but from what I observed in a simple project, is that most build intermediaries will reside in CMakeFiles folder, so this basic implementation will still yield good results for a standard project, even if not perfect. But I haven't test in a real project, so I may get it wrong.

An approach I would like to try in an evolution of rules system, is delegating the cleanup to the project/build tool. For instance, for CMake/make we could detect Makefile and run "make clean" in the appropriate folder. We would lose the reclaim space estimation, but cleanup will be more safe and precise.

**Comment by virtualritz** [comment]
> I don't use CMake daily, but from what I observed in a simple project, is that most build intermediaries will reside in CMakeFiles folder [...]

Yeah, that just contains build *information* files. The actual compiled files are not in there.

To give you an idea: an `OpenSubdiv` debug build is about 159 MB on macOS. 19 MB of that is in various `CMakeFiles` folders scattered in the build. Currently `ocy` only detects the top one which is 0.5 MB. 

I.e. from 159 MB of `CMake` build files, `ocy` frees 0.5 in case of `OpenSubdiv`.

But even if it detected *all* of the `CMakeFiles` folders, it would leave 140 MB of build files undeleted. That's what I mean with "requires parsing [...] to find out what exactly can be deleted".

**Comment by alexd6631** [comment]
Hello @virtualritz,

I got some time to work on the project recently. I removed the current CMake rule which is bogus and in order to support makefile, I added a new kind of rule which allows to run a command inside a dir instead of directly wiping the dir.
While not perfect, It should answer to your need since CMake generate Makefile which support the `make clean` rules.
The biggest downside is that the size estimation is lost in the process, compared to direct deletion.

`ocy 0.1.5` should be up, let me know if it works for you.
